src/runtime/virtcontainers/network_linux.go

```diff
@@ -376,6 +380,99 @@ func (n *LinuxNetwork) addAllEndpoints(ctx context.Context, s *Sandbox, hotplug
 	return nil
 }
 
+func convertDanDeviceToNetworkInfo(device *vctypes.DanDevice) (*NetworkInfo, error) {
+	var netInfo NetworkInfo
+	var err error
+	netInfo.Iface.Name = device.Name
+	if netInfo.Iface.HardwareAddr, err = net.ParseMAC(device.GuestMac); err != nil {
+		return nil, fmt.Errorf("bad mac address in DAN config: %v", err)
+	}
+	netInfo.Iface.MTU = int(device.NetworkInfo.Interface.MTU)
+	netInfo.Iface.Flags = net.Flags(device.NetworkInfo.Interface.Flags)
+
+	for _, addr := range device.NetworkInfo.Interface.IPAddresses {
+		a, err := netlink.ParseAddr(addr)
+		if err != nil {
+			return nil, fmt.Errorf("bad IP address in DAN config: %v", err)
+		}
+
+		netInfo.Addrs = append(netInfo.Addrs, *a)
+	}
+
+	for _, route := range device.NetworkInfo.Routes {
+		var r netlink.Route
+		if len(route.Dest) > 0 {
+			if _, r.Dst, err = net.ParseCIDR(route.Dest); err != nil {
+				return nil, fmt.Errorf("bad route dest in DAN config: %v", err)
+			}
+		}
+		r.Src = net.ParseIP(route.Source)
+		r.Gw = net.ParseIP(route.Gateway)
+		r.Scope = netlink.Scope(route.Scope)
+		if len(r.Gw.To4()) == net.IPv4len {
+			r.Family = unix.AF_INET
+		} else {
+			r.Family = unix.AF_INET6
+		}
+
+		netInfo.Routes = append(netInfo.Routes, r)
+	}
+
+	for _, neigh := range device.NetworkInfo.Neighbors {
+		var n netlink.Neigh
+		n.State = int(neigh.State)
+		n.Flags = int(neigh.Flags)
+		if n.HardwareAddr, err = net.ParseMAC(neigh.HardwareAddr); err != nil {
+			return nil, fmt.Errorf("bad neighbor hardware address in DAN config: %v", err)
+		}
+
+		n.IP = net.ParseIP(neigh.IPAddress)
+		netInfo.Neighbors = append(netInfo.Neighbors, n)
+	}
+
+	return &netInfo, nil
+}
+
+// Load network config in DAN config
+// Create the endpoints for the interfaces in Dan.
+func (n *LinuxNetwork) addDanEndpoints() error {
+
+	jsonData, err := os.ReadFile(n.danConfigPath)
+	if err != nil {
+		return fmt.Errorf("fail to load DAN config file: %v", err)
+	}
+
+	var config vctypes.DanConfig
+	err = json.Unmarshal([]byte(jsonData), &config)
+	if err != nil {
+		return fmt.Errorf("fail to unmarshal DAN config: %v", err)
+	}
+
+	for _, device := range config.Devices {
+		var endpoint Endpoint
+		networkLogger().WithField("interface", device.Name).Info("DAN interface found")
+
+		_, err := convertDanDeviceToNetworkInfo(&device)
+		if err != nil {
+			return err
+		}
+
+		// TODO: Add endpoints that are supported via DAN
+		switch device.Device.Type {
+		default:
+			return fmt.Errorf("unknown DAN device type: '%s'", device.Device.Type)
+		}
+		//nolint: govet
```

Comment:
Why do we need this?

Reply:
This PR doesn't include device supported by DAN, so above `switch...default` always return. 
This is for ignore below lint error:
```
virtcontainers/network_linux.go:466:3: unreachable: unreachable code (govet)
		n.eps = append(n.eps, endpoint)
```

Reply:
Uhh, would you mind adding a todo comment before it? Once implemented, please remove it.

Reply:
sure, will do.